src/modes/exportdata/ExportData.svelte

```diff
@@ -27,18 +27,55 @@
     }
   }
 
+  // Pre-fill the form from URL parameters.
+  // Supported parameters: source, sensor, start, end, geo_type, geo_id
+  const urlParams = new URLSearchParams(window.location.search);
+
+  // Overwrite default source & sensor if these are present in the URL
+  if (urlParams.has('data_source')) {
+    sourceValue = urlParams.get('data_source');
+
+    if (urlParams.has('signal')) {
+      sensorValue = urlParams.get('signal');
+    }
+  }
+
   $: isWeekly = sensor ? sensor.isWeeklySignal : false;
   $: formatDate = isWeekly ? (d) => formatWeek(d).replace('W', '') : formatDateISO;
 
-  let geoType = 'county';
+  let geoType = urlParams.has('geo_type') ? urlParams.get('geo_type') : 'county';
 
   let startDate = new Date();
   let endDate = new Date();
 
   function initDate(date) {
     const param = new DateParam(date);
-    endDate = param.sparkLineTimeFrame.max;
-    startDate = param.sparkLineTimeFrame.min;
+
+    // Populate date based on URL params or, if absent, the current date
+    if (sensor && !sensor.active) {
+      // If the sensor is inactive, set the start and end dates to its historical range rather than current date
+      if (
+        urlParams.has('start_day') &&
```

Comment:
Im not sure we want to clamp these values to the active range if theyve been specified in the URL...  The principle of least surprise suggests showing the user what they asked for.